### PR TITLE
📈 Graphql response time logs

### PR DESCRIPTION
**Pull request by WRadoslaw**
#5729

### CODE REVIEW

File: packages/atlas/src/api/client/index.ts [hunk 1]
```diff
@@ -8,6 +8,29 @@ import { useUserLocationStore } from '@/providers/userLocation'
 
 import { cache } from './cache'
 
+const initializePerformanceObserver = () => {
+  try {
+    const observer = new PerformanceObserver((list) => {
+      for (const entry of list.getEntries()) {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        if ((entry as any).initiatorType === 'fetch') {
```
Comment (zeeshanakram3) [inline packages/atlas/src/api/client/index.ts]:
```suggestion
        if ((entry as PerformanceResourceTiming).initiatorType === 'fetch') {
```

File: packages/atlas/src/api/client/index.ts [hunk 2]
```diff
@@ -8,6 +8,29 @@ import { useUserLocationStore } from '@/providers/userLocation'
...
+          const queryString = entry.name.split('?')?.[1]
+          const params = new URLSearchParams(queryString)
+          const queryType = params.get('queryName')
+          // eslint-disable-next-line no-console
+          console.log(`Query ${queryType ?? entry.name} took ${entry.duration}ms to complete`, entry)
```
Comment (zeeshanakram3) [inline packages/atlas/src/api/client/index.ts]: On the staging env you shared, I am not not able to see specific query types, I see the complete graphql URLs. Is this expected?

image

Comment (zeeshanakram3) [inline packages/atlas/src/api/client/index.ts]: Also, instead of logging, shouldn't these events be sent to Elasticsearch?

**Review by zeeshanakram3 — CHANGES_REQUESTED** [review]
Looks good. It appears that this change will only track Graphql response times, do we also want to track the time taken by REST endpoints as well, e.g YT-Synch, or other services that Atlas might be interacting with.

File: packages/atlas/src/api/client/index.ts (hunk 2 above)
Comment (WRadoslaw) [reply packages/atlas/src/api/client/index.ts]: Yeah, I just logged them for the POC purpose

**Review by zeeshanakram3 — APPROVED** [review]
LGTM

**Pull request merged**